src/components/general/resident-dashboard/DashboardVisualizationProgress.tsx

```diff
@@ -178,9 +191,33 @@ function TargetVariation({ item }: { item: DashboardProgressItem }) {
   );
 }
 
-const DashboardVisualizationProgress = ({ items = [] }: Props) => {
+function getYear(
+  item: DashboardProgressItem,
+  instance: InstanceContextType,
+  scenarioValues: ScenarioValueFieldsFragment[]
+) {
+  if (item.type === ProgressType.SCENARIO) {
+    return scenarioValues.find((scenario) => scenario.scenario.id === item.scenarioId)?.year;
+  }
+
+  if (item.type === ProgressType.CURRENT) {
+    return instance.maximumHistoricalYear;
+  }
+
+  // TODO: Pending backend implementation
```

Comment:
You might be able to dig the year in `instance.goals` bit like in `ScenarioOutcome`, but maybe it's better to get a more robust backend implementation here

Reply:
The problem was that the backend doesn't expose the selected goal index to the UI, but Bernie just made a change to return all goals (year & value) for the selected goal. I'll make another PR for that